primitives/manta/src/util/num.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020-2022 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Numeric Traits
+
+/// Checked Addition
+pub trait CheckedAdd<Rhs = Self> {
```

Comment:
I'm not sure why we have this rust file. Basically, if we derive `AtLeast32BitUnsigned` or `BaseArithmetic`, that will have such methods like `checked_increment`.

Reply:
Ok I see what you mean , why no just use `checked_add(1)` instead of implementing `checked_increment`
@bhgomes did you have something specific in mind here ?

Reply:
@ghzlatarev yeah feel free to use those traits, I didn't know about them at the time of writing the changes